docs/src/main/docs/guides/05_Dockerfile.adoc

```diff
@@ -0,0 +1,302 @@
+     "--add-modules", "java.base,java.logging,java.sql,java.desktop,java.management", \
+     "--output", "/var/tmp/myjre"]
+
+# Work around for https://github.com/docker-library/openjdk/issues/217 <2>
+RUN [ "apt", "update"]
+RUN [ "apt-get", "install", "-y", "binutils"]
+RUN ["strip", "-p", "--strip-unneeded", "/var/tmp/myjre/lib/server/libjvm.so"]
+# End work-around
+
+# Stage 2: Build application image using JRE from Stage 1 <3>
+FROM debian:sid-slim 
+COPY --from=myjre /var/tmp/myjre /opt/jre
+ENV PATH=$PATH:/opt/jre/bin
+
+RUN mkdir /app
+COPY libs /app/libs
+COPY ${project.artifactId}.jar /app
+
+CMD ["java", "-jar", "/app/${project.artifactId}.jar"]
+----
+
+This is a little bit more complicated than our first Dockerfile, in part
+because of a work-around for an openjdk issue. The first thing to notice
+is that this is a multi-stage docker build. That means we're going to
+build multiple docker images -- with later images using content from
+earlier images. 
+
+<1> For the first stage of the build we use a Java 11 base image and we name the image 
+    we are building `myjre`. We then run `jlink` to create a JRE with only the modules
+    we need. We generate that JRE in `/var/tmp/myjre`. The modules listed in this
+    example are for Helidon SE. See below for Helidon MP.
+<2> Ack! We need to work-around an issue in the openjdk base image -- basically run
+    `strip` on `libjvm.so`. Why? Go see https://github.com/docker-library/openjdk/issues/217.
+    After doing that we have a nice shiny new JRE to use.
+<3> Now we build the image for our application. We use `debian:sid-slim` because
+    that matches the base image used by `openjdk:11-slim`, so we can be confident
+    we won't have any runtime compatibility issues with the JRE we created.
+    We copy the JRE from the first image (`myjre`) into our second image, and
+    set our `PATH` so we can find the new JRE. The rest of the file is the same
+    as before.
+
+That's it! You're docker image will now run with a custom JRE. Let's try it:
+
+[source,bash,subs="verbatim,attributes"]
+----
+docker build -t java11-quickstart-se target
+. . .
+docker run --rm -p 8080:8080 java11-quickstart-se:latest
+. . .
+----
+
+The first time you run `docker build` with this Dockerfile it will take a while
+as it downloads stuff and installs `binutils` for the workaround. But subsequent
+runs will be much faster, because all those layers will be cached except for
+the little layer that contains your application jar.
+
+=== What about Helidon MP?
+
+The only difference for Helidon MP is that you need to add a couple more
+modules to the `jlink` command:
+
+[source,yaml,subs="verbatim,attributes"]
+----
+RUN ["jlink", "--compress=2", "--strip-debug", "--no-header-files" \ 
+     "--add-modules", \
+     "java.base,java.logging,java.sql,java.desktop,java.management,java.naming,jdk.unsupported", \
+     "--output", "/var/tmp/myjre"]
+----
+
+=== How Big is This Stuff?
+
```

Comment:
One last time, about separating the commands from their output into separate blocks.

Reply:
Yep